### DISCUSSION TITLE
Weird selection behavior with combobox and gridview using MVUX

### DISCUSSION THREAD

**Discussion by modula2019** [unoplatform/uno#16894] [Q&A]
Hi all, 
I again notice something weird with this combobox and the gridview. I don't know if it is a bug with selection.

Using the combobox, I notice that the selection only gets triggered on the 1st or 2nd attempt. Subsequent selection triggers do not work.
Using the combobox, I also notice that when you set the initial data in the combobox to the below, the selection trigger does not work.
    ` public IState<Person> SelectedPerson => State.Value(this, () => new Person(FirstName: "Master", LastName: "Yoda"));`

 If you do the same for the GridView and you attemt to select the next record, there is an error (***This collection cannot work with indices larger than Int32.MaxValue***).
 
 It seems the selection only wants the initial record to be empty like below. Empty record works perfectly for the GridView but not for the ComboBox.

    `public IState<Person> SelectedPerson => State<Person>.Empty(this); `

 But this is not always the case because I may want to default my initial selection to a particular record number (index) in the list.
 
I also notice when I make changes to the xaml and save, the Hotreload does not work anymore. I am on VS 17.9.6

Thanks for your assistance as usual.

[UnoApp1.zip](https://github.com/unoplatform/uno/files/15469302/UnoApp1.zip)

**Comment by agneszitte** [comment]
cc @dr1rrb
